account_payment_order/wizard/account_payment_line_create.py

```diff
@@ -110,14 +110,18 @@ def _prepare_move_line_domain(self):
             domain += [
                 ('debit', '>', 0),
                 ('account_id.internal_type', 'in', ['receivable', 'payable'])]
-        # Exclude lines that are already in a non-cancelled
-        # and non-uploaded payment order; lines that are in a
-        # uploaded payment order are proposed if they are not reconciled,
-        paylines = self.env['account.payment.line'].search([
-            ('state', 'in', ('draft', 'open', 'generated')),
-            ('move_line_id', '!=', False)])
-        if paylines:
-            move_lines_ids = [payline.move_line_id.id for payline in paylines]
+        # Exclude lines that are already in a non-cancelled payment order
+        self.env.cr.execute(
+            "SELECT aml.id FROM account_payment_line apl "
+            "INNER JOIN account_payment_order apo "
+            "  ON apl.order_id = apo.id "
+            "LEFT OUTER JOIN account_move_line aml "
+            "  ON aml.id = apl.move_line_id "
+            "WHERE apo.state != 'cancel' "
```

Comment:
I think `('state', 'in', ('draft', 'open', 'generated'))`  is not the same as `apo.state != 'cancel'`, and that will probably create performance issues as it will end up in move_line not in "all move lines ever paid".